### PR TITLE
Introduce 'gap' in flexbox Fixes #23221

### LINKED ISSUE #23221: Flexbox guide doesn't include the `gap` property
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Flexible_Box_Layout/Mastering_Wrapping_of_Flex_Items#creating_gutters_between_items
https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Flexible_Box_Layout/Typical_Use_Cases_of_Flexbox#split_navigation

### What specific section or headline is this issue about?

- Creating gutters between items
- Split navigation

### What information was incorrect, unhelpful, or incomplete?

Sections explains old way using negative margins for creating gaps between flexbox items, instead of [`gap`](https://developer.mozilla.org/en-US/docs/Web/CSS/gap):

> When wrapping flex items, the need to space them out is likely to arise. You can see from the live example below that in order to create gaps that do not also create a gap at the edges of the container, we need to use **negative margins on the flex container itself**. Any border on the flex container is then moved to a second wrapper in order that the negative margin can pull the items up to that wrapper element.
It is this requirement **that the `gap` properties, once implemented, will solve for us**. Proper gaps only happen on the inside edges of items.

And:

> Also in this example we are using margins on the flex items to create a gap between items, and a **negative margin on the container** in order that the items still remain flush with the right and left edges. **Until the `gap` properties from the Box Alignment specification are implemented** in flexbox, we have to use margins in this way if we want to create a gutter between items.

### What did you expect to see?

Update instructions to use `gap`.

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/css/css_flexible_box_layout/mastering_wrapping_of_flex_items`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Flexible_Box_Layout/Mastering_Wrapping_of_Flex_Items
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/css/css_flexible_box_layout/mastering_wrapping_of_flex_items/index.md
* Last commit: https://github.com/mdn/content/commit/71c4bc0b6329ec40ddbefd8d3124547e91cfa612
* Document last modified: 2022-09-28T03:40:57.000Z

</details>

### PR DESCRIPTION
**Pull request by MAHIN0093**
### Description
Adds a subsection about the `gap` properties in Flexbox

### Motivation
Creating negative margins can be a nightmare and new learners would be benefited more with 'gap' being introduced.

### Related issues and pull requests
Fixes #23221

### CODE REVIEW

**Comment by wbamberg** [comment]
Thanks for your PR!

But I think we need a more extensive update here. This whole section should be about `gap`, and only talk about negative margins as an afterthought (or better perhaps, not talk about them at all). That means we also need an update to the example.

Please let me know if you still want to take this issue or not.

Also, please be careful with writing: spaces after periods, and no double spaces.

**Comment by MAHIN0093** [comment]
> Thanks for your PR!
> 
> But I think we need a more extensive update here. This whole section should be about `gap`, and only talk about negative margins as an afterthought (or better perhaps, not talk about them at all). That means we also need an update to the example.
> 
> Please let me know if you still want to take this issue or not.
> 
> Also, please be careful with writing: spaces after periods, and no double spaces.

Yes. Assign this one to me.

File: files/en-us/web/css/css_flexible_box_layout/mastering_wrapping_of_flex_items/index.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ If you need flex items to line up in the cross axis, controlling the width in th
 
 When wrapping flex items, the need to space them out is likely to arise. You can see from the live example below that in order to create gaps that do not also create a gap at the edges of the container, we need to use negative margins on the flex container itself. Any border on the flex container is then moved to a second wrapper in order that the negative margin can pull the items up to that wrapper element.
 
-It is this requirement that the gap properties, once implemented, will solve for us. Proper gaps only happen on the inside edges of items.
+A better way to perform the above tasks would be to use the Flexbox `gap` property.It is specified on the container and creates a equal space among all flex items.Try to implement gaps in the  both using negative margins and the `gap` property in the editor below.
```
Comment (estelle) [inline files/en-us/web/css/css_flexible_box_layout/mastering_wrapping_of_flex_items/index.md:65]:
```suggestion
A better way to perform the above tasks is to include the Flexbox `gap` property on the container to set equal space between all flex items. Try setting gaps using both negative margins and the `gap` property in the editor below.
```

**Review by estelle — APPROVED** [review]
Thank you.

**Comment by estelle** [comment]
Apologies. I was reviewing at the same time that you two were having the conversation, and didn't see it before merging. I think adding the addition of the gap property you suggested is helpful until the more extensive edits are made, but can revert if you prefer.

**Comment by wbamberg** [comment]
No worries. I think it is better with this edit so let's keep it. But I don't think we can really call this a fix for the original issue, so I reopened that.

**Pull request merged**